gt/main.c

```diff
@@ -1770,3 +1839,46 @@ run_gt(struct net_config *net_conf, struct gt_config *gt_conf)
 	rte_atomic32_init(&gt_conf->ref_cnt);
 	return 0;
 }
+
+void
+update_lua_states(struct gt_config *gt_conf)
+{
+	int i;
+
+	for (i = 0; i < gt_conf->num_lcores; i++) {
+		int ret;
+		struct gt_cmd_entry *entry;
+		struct gt_instance *instance = &gt_conf->instances[i];
+		lua_State *lua_state = alloc_and_setup_lua_state();
+		if (lua_state == NULL) {
+			RTE_LOG(WARNING, GATEKEEPER,
+				"gt: failed to allocate new lua state to GT block %u at lcore %u\n",
+				i, rte_lcore_id());
```

Comment:
We cannot call `rte_lcore_id()` here because the caller of `update_lua_states()` is the Dynamic configuration block, not a GT block. We need to use `gt_conf->lcores[i]` instead.